### PR TITLE
Lowess

### PR DESCRIPTION
**Pull request by koaning**
I like the implementation as far as API goes. 

image

### CODE REVIEW

**Comment by koaning** [comment]
But the performance is really slow. Note the effect of the span parameter. 

image

**Comment by MBrouns** [comment]
The implementation looks good! There is a minor inconsistency in the Wording of the parameter validation exceptions though

**Comment by koaning** [comment]
Could you rephrase what you mean with "Wording of the paramater validation exceptions"?

**Comment by MBrouns** [comment]
Ah sure, 

```
if self.span is not None:
            if not 0 <= self.span <= 1:
                raise ValueError(f"we must have 0 <= span <= 1, got: {self.span}")
        if self.sigma < 0:
            raise ValueError(f"sigma must be >= 0, got: {self.sigma}")
```

The errors are phrased slightly differently.

**Comment by koaning** [comment]
I completely forgot about this branch. May wrap it up tonight.

**Comment by MBrouns** [comment]
this looks almost finished right? Just need to add tests?

**Comment by koaning** [comment]
And docs. But this is something I'd love to pick up today.

**Comment by koaning** [comment]
The docs will get fancy gifs. 

image

**Comment by koaning** [comment]
I think this is it. Once we merge this we'll also need to push to pypi though because the docs are going to list a new method.
